File: app/src/main/java/com/dublikunt/nclient/utility/LogUtility.java
package com.dublikunt.nclient.utility;

import android.util.Log;

import java.util.Arrays;

public class LogUtility {
    public static final String LogTag = "NCLIENT";

    public static void d(Object... message) {
        if (message == null) return;
        if (message.length == 1) Log.d(LogUtility.LogTag, "" + message[0]);
        else Log.d(LogUtility.LogTag, Arrays.toString(message));
    }

    public static void d(Object message, Throwable throwable) {
        if (message == null) message = "";
        Log.d(LogUtility.LogTag, message.toString(), throwable);
    }

    public static void i(Object... message) {
        if (message == null) return;
        if (message.length == 1) Log.i(LogUtility.LogTag, "" + message[0]);
        else Log.i(LogUtility.LogTag, Arrays.toString(message));
    }

    public static void i(Object message, Throwable throwable) {
        if (message == null) message = "";
        Log.i(LogUtility.LogTag, message.toString(), throwable);
    }

    public static void e(Object... message) {
        if (message == null) return;
        if (message.length == 1) Log.e(LogUtility.LogTag, "" + message[0]);
        else Log.e(LogUtility.LogTag, Arrays.toString(message));
    }

    public static void e(Object message, Throwable throwable) {
        if (message == null) message = "";
        Log.e(LogUtility.LogTag, message.toString(), throwable);
    }
}

File: app/src/main/kotlin/com/dublikunt/nclient/utility/LogUtility.kt
package com.dublikunt.nclient.utility

import android.util.Log

object LogUtility {
    private const val LOGTAG = "NCLIENTLOG"

    @JvmStatic
    fun download(vararg message: Any) {
        if (message.size == 1) Log.d(LOGTAG, "" + message[0]) else Log.d(
            LOGTAG, message.contentToString()
        )
    }

    @JvmStatic
    fun download(message: String?, throwable: Throwable?) {
        var message = message
        if (message == null) message = ""
        Log.d(LOGTAG, message, throwable)
    }

    @JvmStatic
    fun info(vararg message: Any) {
        if (message.size == 1) Log.i(LOGTAG, "" + message[0]) else Log.i(
            LOGTAG, message.contentToString()
        )
    }

    @JvmStatic
    fun info(message: String?, throwable: Throwable?) {
        var message = message
        if (message == null) message = ""
        Log.i(LOGTAG, message, throwable)
    }

    @JvmStatic
    fun error(vararg message: Any) {
        if (message.size == 1) Log.e(LOGTAG, "" + message[0]) else Log.e(
            LOGTAG, message.contentToString()
        )
    }


    @JvmStatic
    fun error(message: Any?, throwable: Throwable?) {
        var message = message
        if (message == null) message = ""
        Log.e(LOGTAG, message.toString(), throwable)
    }
}
